File: application/frontend/src/app/model/Product.ts
export class Product {
    public id: number | null = null;
    public name: string = "";
    public price: number = -1;
    public description: string = "";
    public category: string = "";
    public userId: number | null = null;

    public constructor(
        id: number | null,
        name: string,
        description: string,
        category: string,
        userId: number | null,
    ) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.category = category;
        this.userId = userId
    }
}

File: application/backend/src/main/java/com/auctix/auctx/model/Product.java
package com.auctix.auctx.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import org.hibernate.validator.constraints.Length;

import javax.persistence.*;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "product")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(name = "name")
    @Length(min = 6, max = 200)
    private String name;

    @Column(name = "description")
    @Length(min = 6, max = 1000)
    private String description;

    @Column(name = "price")
    private Float price;

    @Column(name = "category")
    @Enumerated(EnumType.STRING)
    private Category category;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    @ToString.Exclude
    @JsonIgnore
    private Users user;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
    @ToString.Exclude
    @JsonIgnore
    private List<ProductReview> reviews;

}
